stdlib/messaging/activemq-artemis/src/main/ballerina/artemis/producer.bal

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents ActiveMQ Artemis Producer
+public type Producer client object {
+    private Session session;
+    private boolean anonymousSession = false;
+
+    public function __init(Session | URLConfiguration sesssionOrURLConfig, string addressName, AddressConfiguration? addressConfig = (), int rate = -1) {
+        if (sesssionOrURLConfig is Session) {
+            self.session = sesssionOrURLConfig;
+        } else {
+            //Todo: url parse host and the port
+            Connection connection = new("tcp://" + sesssionOrURLConfig.host + ":" + sesssionOrURLConfig.port);
+            self.session = new(connection, config = {username: sesssionOrURLConfig.username, password: sesssionOrURLConfig.password});
+            self.anonymousSession = true;
+        }
+        AddressConfiguration configuration = {};
+        if (addressConfig is AddressConfiguration) {
+            configuration = addressConfig;
+        }
+        self.createProducer(addressName, configuration, rate);
+    }
+
+    extern function createProducer(string addressName, AddressConfiguration addressConfig, int rate);
+
+    # Sends a message to the producer's address
+    #
+    # + data - the `Message` or data to send 
+    # + return - `error` on failure
+    public remote function send(int | float | string | json | xml | byte | byte[] | map<string | int | float | byte
+| boolean | byte[]> | io:ReadableByteChannel | Message data) returns error? {
+        Message msg;
```

Comment:
```suggestion
        Message msg = data is Message ? data : new(self.session, data);
```

Reply:
This conversion is not possible because `data` is not recognized as type Message inside the ternary statement

Reply:
Can you share the error please, this should work IINM? Tried with the following simple example:
```ballerina
import ballerina/io;

public function main(string... args) {
    string|int|Message m = "hello";
    testTernary(m);
}

function testTernary(string|int|Message m) {
    Message m2 = m is Message ? m : new Message(m);
    m2->printText();    
}

public type Message client object {
    string|int text;

    public function __init(string|int s) {
        self.text = s;
    }

    remote function printText() {
        io:println(self.text);
    }
};
```

Reply:
That seems to work. I have been mistaken. Thanks..